projects/valtimo/components/src/lib/services/cds-theme.service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// theme-preference.service.ts
+import {Injectable, OnDestroy} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subscription} from 'rxjs';
+import {SelectableCarbonTheme} from '../models';
+import {filter} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CdsThemeService implements OnDestroy {
+  private readonly _preferredTheme$ = new BehaviorSubject<SelectableCarbonTheme>(
+    SelectableCarbonTheme.G10
+  );
+
+  public get preferredTheme$(): Observable<string> {
+    return this._preferredTheme$.pipe(filter(theme => !!theme));
+  }
+
+  private readonly _subscriptions = new Subscription();
+
+  constructor() {
+    this.openDarkModeSubscription();
+  }
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public setPreferredTheme(selectedTheme: SelectableCarbonTheme): void {
+    if (selectedTheme) this._preferredTheme$.next(selectedTheme);
+  }
```

Comment:
```suggestion
  public setPreferredTheme(selectedTheme: SelectableCarbonTheme | undefined): void {
    if (selectedTheme) {
        this._preferredTheme$.next(selectedTheme);
     }
  }
```

Reply:
I think the former looks cleaner, plus we have no style guide preferring one or the other